Jenkinsfile

```diff
@@ -947,7 +947,9 @@ def withBeatsEnv(Map args = [:], Closure body) {
         if (archive) {
           archiveTestOutput(testResults: '**/build/TEST*.xml', artifacts: '**/build/TEST*.out')
         }
-        reportCoverage()
+        catchError(buildResult: 'SUCCESS', stageResult: 'UNSTABLE') {
+          sh(label: 'Report to Codecov', script: '.ci/scripts/report-codecov.sh auditbeat filebeat heartbeat journalbeat libbeat metricbeat packetbeat winlogbeat')
```

Comment:
Why not use the codecov step from the library?

Reply:
Uhm, that's a good point, my aim was to add more resilience to the 4xx http errors when curling the codecov... so I'll iterate a few times and see if we can simplify the pipeline to use the steps as you said

Reply:
I've just raised an issue https://github.com/elastic/beats/issues/20080 to fix the the codecov issues in the pipeline since the credentials are not in place as far as I see, so for the time being I'll keep this PR as is and I'll work on the issue afterwards